src/irmin-pack/unix/pack_store.ml

```diff
@@ -420,6 +420,14 @@ struct
     Lwt.try_bind (fun () -> f (cast t)) on_success on_fail
 
   let unsafe_append ~ensure_unique ~overcommit t hash v =
+    let kind = Val.kind v in
+    let use_index =
+      (* the index is needed for non-minimal indexing strategies and
+         for commits in the v1 format (as these never use direct
+         addressing) *)
+      (not (Irmin_pack.Indexing_strategy.is_minimal t.indexing_strategy))
+      || kind = Commit_v1
```

Comment:
shouldn't it be `Commit_v2`?

Reply:
Do we need to use the index for Commit_v2? I though it was only useful for v1 and v2 was always using offsets?

Reply:
My understanding is that commit_v1 are no longer generated with irmin >= 3.0 and that 
commit_v2 are indexed anyway in both minimal and always. Or do you mean that for commit_v2 in minimal we can just append them again because we anyway append all its tree?  

I'm probably missing something

Reply:
I'm probably missing something too :-) I've added both commit versions so that the index is always used for commit objects.